onnxruntime/contrib_ops/cpu/word_conv_embedding.cc

```diff
@@ -0,0 +1,225 @@
+#include "word_conv_embedding.h"
+
+#include "core/util/math.h"
+#include "core/util/math_cpuonly.h"
+#include "core/mlas/inc/mlas.h"
+
+namespace onnxruntime {
+namespace contrib {
+
+void WordConvEmbedding::CharEmbeddingLookup(
+    const int* seq_ptr,
+    const float* char_embedding_weight_p,
+    size_t seq_len,
+    size_t word_len,
+    size_t char_embedding_size,
+    const int* words_len_ptr,
+    float* dst) const {
+  for (size_t word_inx = 0; word_inx < seq_len; word_inx++) {
+    if (words_len_ptr[word_inx] > 0) {
+      const int* cur_seq_ptr = seq_ptr + word_inx * word_len;
+      float* cur_dst_ptr = dst + word_inx * word_len * char_embedding_size;
+      for (size_t char_inx = 0; char_inx < word_len; char_inx++) {
+        memcpy(cur_dst_ptr, char_embedding_weight_p + (*cur_seq_ptr) * char_embedding_size, sizeof(float) * char_embedding_size);
+        cur_dst_ptr += char_embedding_size;
+        cur_seq_ptr++;
+      }
+    }
+  }
+}
+
+//input : [batch, sequence_length, word_length, char_embedding_size]
+void WordConvEmbedding::ComputeConvMaxPoolWithActivation(
+    AllocatorPtr allocator,
+    const float* input,
+    const float* weights,
+    const float* bias,
+    const int* words_len_ptr,
+    int64_t seq_len,
+    int64_t word_len,
+    int64_t char_embedding_size,
+    int64_t filter_width,
+    int64_t num_filters,
+    float* output) const {
+  int64_t input_word_size = word_len * char_embedding_size;
+  int64_t unfolded_width = word_len - filter_width + 1;
+  int64_t unfolded_kernal_size = filter_width * char_embedding_size;
+  int64_t unfolded_segment_size = unfolded_width * unfolded_kernal_size;
+  //int conv_res_segment_size = _filter_width * _num_filters; // TODO: it imay be unfolded_width * _num_filters
```

Comment:
// TODO: it **imay** be unfolded_width * _num_filters
Typo?

Reply:
Don't need the comment any more.